### PR TITLE
[NETBEANS-5686] Upgrade module 'websvc.restlib' libraries

### PR DESCRIPTION
**Pull request by pepness**
Notes:

- Reduced size of jaxrs/jersey library by 20%

| Directory | Size before | Size after |
| --------------- | --------------- | --------------- |
| jaxrs-2.0/jaxrs-2.1 | 0.1128 MB  | 0.1404 MB |
| jersey2 | 6.1 MB | 4.8 MB |
| Total | 6.2128 MB | 4.9404 MB |
- Utilize jakarta libraries instead of javax
- Update license type to EPL-v20 where needed
- Upgrade hk2 from 2.2 to 2.6.1
 -- Add aopalliance-repackaged dependency
 -- Remove cglib dependency
 -- Remove asm-all-repackaged dependency (2.5.0-b64)
- Upgrade jersey from 2.5.1 to 2.34
 -- Remove guava dependency
- Use one license file for all jersey libraries and use license type EPL-v20
- Upgrade versions inside Bundle.properties file
- Fix validation inside ignored-overlaps file
- Use one license file for hk2 dependencies

Testing:
[Project](https://github.com/pepness/jerseyapp) to test this PR
- Full build done
- Verify successfull execution of libraries and licenses Ant test
- Verify successful execution of unit tests for module `websvc.restlib`
- Checked the files are in the Ant Library Manager
- Successfully create project as per [Jersey documentation](https://eclipse-ee4j.github.io/jersey.github.io/documentation/latest/getting-started.html), with Apache Tomcat and TomEE
- Ant based project based on Jersey documentation in https://github.com/pepness/jerseyapp
- Successfully create project as per [NetBeans Documentation](https://netbeans.apache.org/kb/docs/websvc/rest.html), with Payara and Glassfish
- Successfully create project as per [Oracle Documentation](https://www.oracle.com/webfolder/technetwork/tutorials/obe/java/RESTfulWebServices/RESTfulWebservices.htm), with Payara and Glassfish

Will do final squash if needed by the RM.

### CODE REVIEW

**Comment by pepness** [comment]
Travis failed because of timeouts but I think is ready to be merged. Can someone have a look? @ebarboni @matthiasblaesing @sdedic

**Comment by ebarboni** [comment]
seems ok rerun is sometimes working :D. maybe @jGauravGupta as I saw payara may help reviewing this

**Comment by jGauravGupta** [comment]
+1 LGTM, Tested Maven/Ant application on Payara and GlassFish Server.

File: nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps [hunk 1]
```diff
@@ -45,14 +45,19 @@ nbbuild/external/json-simple-1.1.1.jar ide/libs.json_simple/external/json-simple
 webcommon/javascript2.jade/external/jflex-1.4.3.jar webcommon/javascript2.lexer/external/jflex-1.4.3.jar
 
 # javax.annotation-api is used by multiple modules.
-enterprise/websvc.restlib/external/javax.annotation-api-1.2.jar java/maven.indexer/external/javax.annotation-api-1.2.jar
 enterprise/javaee.api/external/javax.annotation-api-1.2.jar java/maven.indexer/external/javax.annotation-api-1.2.jar
```
Comment (jGauravGupta) [inline nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps:48]: javax.annotation-api-1.2 upgrade to jakarta.annotation-api:1.3.5

Comment (pepness) [reply nbbuild/antsrc/org/netbeans/nbbuild/extlibs/ignored-overlaps:48]: I upgraded the javax.annotation-api only inside the `enterprise.websvc` module. In [PR-1809](https://github.com/apache/netbeans/pull/1809) Matthias Bläsing stated that I should only use the versions that belong to each release of JavaEE. I think in this case I should create another PR to upgrade the version of javax.annotation-api for the modules `enterprise/javaee8.api` to javax.annotation-api:1.3.2 and `java/maven.indexer` to jakarta.annotation-api:1.3.5. I hope I understood your comment.

**Review by jGauravGupta — APPROVED** [review]

**Comment by pepness** [comment]
Can this make it to 12.5 @ebarboni?

**Comment by ebarboni** [comment]
@pepness I think so

**Comment by pepness** [comment]
Should I squash the commits @ebarboni?

**Comment by ebarboni** [comment]
I would say yes

**Pull request merged**